### PR TITLE
Port sunSunSafePoint to C++

### LINKED ISSUE #196: Port sunSafePoint module to C++
**Describe your use case**
The sunSafePoint module will be ported to C++. The old module name is unchanged; while the new module is called `sunSafePointCpp`.

**Describe alternatives solutions you've considered**
N/A

**Additional context**
N/A

### PR DESCRIPTION
**Pull request by leahkiner**
* **Tickets addressed:** bsk-758
* **Review:** By commit
* **Merge strategy:** Merge (no squash)

## Description
This PR creates a C++ version of the `sunSafePoint` fsw module. The module functionality is not changed.

## Verification
The old unit test script is kept and the new module passes the test.

## Documentation
N/A

## Future work
N/A

### CODE REVIEW

**Comment by leahkiner** [comment]
@patkenneally I just made a bunch of changes to this branch to reflect your comments on the [mrpProportionalDerivative](https://github.com/lasp/basilisk/pull/194) PR. This PR is now ready for your review!

File: src/fswAlgorithms/attGuidance/sunSafePointCpp/sunSafePointCpp.cpp
Comment (patkenneally) [inline src/fswAlgorithms/attGuidance/sunSafePointCpp/sunSafePointCpp.cpp]: :small_red_triangle_down: We should add `const` to each of 

`void SunSafePointCpp::setMinUnitMag(double minUnitMag) const`
`void SunSafePointCpp::setSmallAngle(double smallAngle) const`
`void SunSafePointCpp::setSunAxisSpinRate(double sunAxisSpinRate) const`

File: src/fswAlgorithms/attGuidance/sunSafePointCpp/sunSafePointCpp.cpp
Comment (patkenneally) [inline src/fswAlgorithms/attGuidance/sunSafePointCpp/sunSafePointCpp.cpp]: :small_red_triangle_down: We can make the reference to the parameter in these `const`
`void SunSafePointCpp::setOmega_RN_B(const Eigen::Vector3d &omega_RN_B)`
`void SunSafePointCpp::setSHatBdyCmd(const Eigen::Vector3d &sHatBdyCmd) `

File: src/fswAlgorithms/attGuidance/sunSafePointCpp/sunSafePointCpp.h
Comment (patkenneally) [inline src/fswAlgorithms/attGuidance/sunSafePointCpp/sunSafePointCpp.h]: :large_orange_diamond: We can delete this and the comment below. They are redundant because we've now indicated clear what is available to the user on the public API of this module.

File: src/fswAlgorithms/attGuidance/sunSafePointCpp/sunSafePointCpp.cpp
Comment (patkenneally) [inline src/fswAlgorithms/attGuidance/sunSafePointCpp/sunSafePointCpp.cpp:99]: :large_orange_diamond: There's a funky new line formatting here.

File: src/fswAlgorithms/attGuidance/sunSafePointCpp/sunSafePointCpp.cpp
Comment (patkenneally) [inline src/fswAlgorithms/attGuidance/sunSafePointCpp/sunSafePointCpp.cpp]: :large_orange_diamond: We should declare this variable within the scope that it is used. This would be inside the 'else' clause.

File: src/fswAlgorithms/attGuidance/sunSafePointCpp/sunSafePointCpp.cpp
Comment (patkenneally) [inline src/fswAlgorithms/attGuidance/sunSafePointCpp/sunSafePointCpp.cpp]: :large_blue_circle: We can declare and initialize message variables on a single line. E.g. `NavAttMsgPayload localImuDataInBuffer =  NavAttMsgPayload();`

File: src/fswAlgorithms/attGuidance/sunSafePointCpp/_UnitTest/test_sunSafePointCpp.py
Comment (patkenneally) [inline src/fswAlgorithms/attGuidance/sunSafePointCpp/_UnitTest/test_sunSafePointCpp.py:167]: Thank you!

**Review by patkenneally — CHANGES_REQUESTED** [review]
Legend:
:small_red_triangle_down: *Issues* to address before merge
:large_orange_diamond: *Requests* that should not block merge, but should at least be discussed
:large_blue_circle: *Recommendations* that can be ignored if desired

File: src/fswAlgorithms/attGuidance/sunSafePointCpp/sunSafePointCpp.h
Comment (leahkiner) [reply src/fswAlgorithms/attGuidance/sunSafePointCpp/sunSafePointCpp.h]: Done

File: src/fswAlgorithms/attGuidance/sunSafePointCpp/sunSafePointCpp.cpp
Comment (leahkiner) [reply src/fswAlgorithms/attGuidance/sunSafePointCpp/sunSafePointCpp.cpp]: Done

File: src/fswAlgorithms/attGuidance/sunSafePointCpp/sunSafePointCpp.cpp
Comment (leahkiner) [reply src/fswAlgorithms/attGuidance/sunSafePointCpp/sunSafePointCpp.cpp]: Done

File: src/fswAlgorithms/attGuidance/sunSafePointCpp/sunSafePointCpp.cpp
Comment (leahkiner) [reply src/fswAlgorithms/attGuidance/sunSafePointCpp/sunSafePointCpp.cpp]: Done

File: src/fswAlgorithms/attGuidance/sunSafePointCpp/sunSafePointCpp.cpp
Comment (leahkiner) [reply src/fswAlgorithms/attGuidance/sunSafePointCpp/sunSafePointCpp.cpp:99]: I moved the division symbol to line up with the `t` in `this->sHatBdyCmd`

File: src/fswAlgorithms/attGuidance/sunSafePointCpp/sunSafePointCpp.cpp
Comment (leahkiner) [reply src/fswAlgorithms/attGuidance/sunSafePointCpp/sunSafePointCpp.cpp]: Done

File: src/fswAlgorithms/attGuidance/sunSafePointCpp/sunSafePointCpp.cpp
Comment (patkenneally) [reply src/fswAlgorithms/attGuidance/sunSafePointCpp/sunSafePointCpp.cpp]: Sorry. My bad. It's the get functions that can be marked `const`, not the setters as you correctly queried.

Comment (leahkiner) [reply src/fswAlgorithms/attGuidance/sunSafePointCpp/sunSafePointCpp.cpp]: Done!

**Review by patkenneally — APPROVED** [review]

**Pull request merged**
